File: CSharpModelTrainerApi/LungCancerPrediction/Services/TrainingHelper.cs
using TorchSharp;
using static TorchSharp.torch;

namespace CSharpModelTrainerApi.LungCancerPrediction.Services
{
    public class TrainingHelper
    {
        public static torch.Device GetOptimalDevice()
        {
            if (torch.cuda_is_available())
            {
                return torch.CUDA;
            }
            else
            {
                return torch.CPU;
            }
        }
    }
}

File: python-model-trainer/LungCancerPrediction/Services/training_helper.py

import torch

class TrainingHelper:
    @staticmethod
    def get_optimal_device() -> torch.device:
        if torch.cuda.is_available():
            return torch.device("cuda:0")
        elif torch.backends.mps.is_available():
            return torch.device("mps:0")
        else:
            return torch.device("cpu")